diff -Naur smesh-5.1.2.2.orig/src/Controls/SMESH_Controls.cpp smesh-5.1.2.2/src/Controls/SMESH_Controls.cpp
--- smesh-5.1.2.2.orig/src/Controls/SMESH_Controls.cpp	2009-12-16 09:01:21.000000000 -0600
+++ smesh-5.1.2.2/src/Controls/SMESH_Controls.cpp	2013-02-15 14:58:28.357147721 -0600
@@ -64,6 +64,11 @@
 #include "SMESHDS_Mesh.hxx"
 #include "SMESHDS_GroupBase.hxx"
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif 
+
 /*
                             AUXILIARY METHODS
 */
diff -Naur smesh-5.1.2.2.orig/src/SMESH/SMESH_MeshEditor.cpp smesh-5.1.2.2/src/SMESH/SMESH_MeshEditor.cpp
--- smesh-5.1.2.2.orig/src/SMESH/SMESH_MeshEditor.cpp	2013-02-15 14:42:44.453062739 -0600
+++ smesh-5.1.2.2/src/SMESH/SMESH_MeshEditor.cpp	2013-02-15 14:51:15.754435891 -0600
@@ -84,6 +84,10 @@
 
 #define cast2Node(elem) static_cast<const SMDS_MeshNode*>( elem )
 
+#ifndef PI 
+# define PI M_PI 
+#endif 
+
 using namespace std;
 using namespace SMESH::Controls;
 
diff -Naur smesh-5.1.2.2.orig/src/SMESH/SMESH_Pattern.cpp smesh-5.1.2.2/src/SMESH/SMESH_Pattern.cpp
--- smesh-5.1.2.2.orig/src/SMESH/SMESH_Pattern.cpp	2013-02-15 14:42:44.457062609 -0600
+++ smesh-5.1.2.2/src/SMESH/SMESH_Pattern.cpp	2013-02-15 14:55:31.929960525 -0600
@@ -75,6 +75,11 @@
 
 #include "utilities.h"
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif 
+
 using namespace std;
 
 typedef map< const SMDS_MeshElement*, int > TNodePointIDMap;
diff -Naur smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_AutomaticLength.cpp smesh-5.1.2.2/src/StdMeshers/StdMeshers_AutomaticLength.cpp
--- smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_AutomaticLength.cpp	2009-12-16 09:01:21.000000000 -0600
+++ smesh-5.1.2.2/src/StdMeshers/StdMeshers_AutomaticLength.cpp	2013-02-15 15:00:49.259529380 -0600
@@ -38,6 +38,11 @@
 #include <TopoDS.hxx>
 #include <TopoDS_Edge.hxx>
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif 
+
 using namespace std;
 
 //=============================================================================
diff -Naur smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_CompositeHexa_3D.cpp smesh-5.1.2.2/src/StdMeshers/StdMeshers_CompositeHexa_3D.cpp
--- smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_CompositeHexa_3D.cpp	2009-12-16 09:01:21.000000000 -0600
+++ smesh-5.1.2.2/src/StdMeshers/StdMeshers_CompositeHexa_3D.cpp	2013-02-15 15:03:10.805901650 -0600
@@ -57,6 +57,10 @@
 #include <set>
 #include <vector>
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif 
 
 #ifdef _DEBUG_
 
diff -Naur smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_QuadToTriaAdaptor.cpp smesh-5.1.2.2/src/StdMeshers/StdMeshers_QuadToTriaAdaptor.cpp
--- smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_QuadToTriaAdaptor.cpp	2009-12-16 09:01:21.000000000 -0600
+++ smesh-5.1.2.2/src/StdMeshers/StdMeshers_QuadToTriaAdaptor.cpp	2013-02-15 15:05:12.937914205 -0600
@@ -40,6 +40,11 @@
 #include <gp_Lin.hxx>
 #include <gp_Pln.hxx>
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif
+
 #ifndef __BORLANDC__
 #include <NCollection_Array1.hxx>
 typedef NCollection_Array1<TColStd_SequenceOfInteger> StdMeshers_Array1OfSequenceOfInteger;
diff -Naur smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_RadialQuadrangle_1D2D.cpp smesh-5.1.2.2/src/StdMeshers/StdMeshers_RadialQuadrangle_1D2D.cpp
--- smesh-5.1.2.2.orig/src/StdMeshers/StdMeshers_RadialQuadrangle_1D2D.cpp	2009-12-16 09:01:21.000000000 -0600
+++ smesh-5.1.2.2/src/StdMeshers/StdMeshers_RadialQuadrangle_1D2D.cpp	2013-02-15 15:09:28.717572007 -0600
@@ -55,6 +55,10 @@
 #include <TopExp_Explorer.hxx>
 #include <TopoDS.hxx>
 
+#ifndef PI 
+# include <math.h> 
+# define PI M_PI 
+#endif
 
 using namespace std;
 
